matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt

```diff
@@ -47,7 +49,13 @@ internal class RoomMemberEventHandler @Inject constructor(
         if (roomMember == null) {
             return false
         }
-        val roomMemberEntity = RoomMemberEntityFactory.create(roomId, userId, roomMember)
+        val roomMemberEntity = RoomMemberEntityFactory.create(
+                roomId,
+                userId,
+                roomMember,
+                // To resolve an edge cases like, on initial sync RoomMemberEventHandler is called after PresenceSyncHandler,
```

Comment:
maybe we can change and call PresenceSyncHandler after RoomMemberEventHandler?

Reply:
Well, I deep dived into the problem about what is happening. I will also add a comment in the codebase explaining that. Let me explain, when we use `insertOrUpdate` then we have two results:

1. **Insert** : A brand new object is created with a primary key etc and is inserted in the DB
2. **Update**: A primary key is already found in that table so the framework **replace** the existing object with the new one. 

The problem is when there is an **update**! With An example will be more clear. Let's say we have the following record:
[ **MyPrimaryKey**: 123, catName: "Anna", dogName:"John" , .... etc]

and now lets say server has an update with [ **MyPrimaryKey**: 123, catName: "Ketty"]

So now if we do `insertOrUpdate` with an object containing only **MyPrimaryKey** and **catName**. All the other records will be replaced by null, so after the **insertOrUpdate** our record will be updated to:
[ **MyPrimaryKey**: 123, catName: "Anna", dogName:**NULL** , .... etc]

So to avoid this there are two solutions:

1.  Manually check if the record exists, and update only the fields we want to update
2. Pick the old value and pass it to new object creation (Faster if the select is with **@indexed** fields)

While **userId** and **roomId** are **@indexed**, I think the second one is clearer and more efficient. What do you think?